server/src/language/models/cache.js

```diff
@@ -42,6 +42,10 @@ export default class Cache {
 
     /** @type {IncludeStatement[]} */
     this.includes = cache.includes || [];
+
+    /** @type {Declaration[]} */
+    this.cursor = cache.cursor || [];
```

Comment:
Please use a plural here. `cursor` to be `cursors` to follow the rest of the properties in this class.